src/lib/metrics.rs

```diff
@@ -457,6 +458,19 @@ impl DemuxedGroupSampleMetrics {
                 / self.base_qual_counter.index_bases_total_seen as f64,
         }
     }
+
+    fn get_metrics_barcode(sample_metadata: &SampleMetadata) -> String {
```

Comment:
1. might as well add some function documentation
2. why not make this a method on `SampleMetadata`?

Reply:
Moved. Please re-review. Note that moving this to `SampleMetadata` highlights how we should be better about validating the struct.